### PR TITLE
 kubeadm: Remove the `.CloudProvider` and `.PrivilegedPods` configuration option

### PR DESCRIPTION
**Pull request by luxas**
**What this PR does / why we need it**:
Removes the `.CloudProvider` option, it has been experimental for a long time. People should now use external cloud providers, which is beta in v1.11. Most importantly, you can get the exact same behavior in the API by utilizing the `.*ExtraArgs` and `.*ExtraVolumes` fields.
Removes `.PrivilegedPods` as that serves a super small edge case with the legacy cloud provider, and only for openstack.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Part of https://github.com/kubernetes/community/pull/2131

**Special notes for your reviewer**:
Depends on PRs:
 - [x] #63799
 - [x] #63788

**Release note**:

```release-note
[action required] In the new v1alpha2 kubeadm Configuration API, the `.CloudProvider` and `.PrivilegedPods` fields don't exist anymore.
Instead, you should use the out-of-tree cloud provider implementations which are beta in v1.11.
If you have to use the legacy in-tree cloud providers, you can rearrange your config like the example below. In case you need the `cloud-config` file (located in `{cloud-config-path}`), you can mount it into the API Server and controller-manager containers using ExtraVolumes like the example below.
If you need to use the `.PrivilegedPods` functionality, you can still edit the manifests in
`/etc/kubernetes/manifests/`, and set `.SecurityContext.Privileged=true` for the apiserver
and controller manager.
---
kind: MasterConfiguration
apiVersion: kubeadm.k8s.io/v1alpha2
apiServerExtraArgs:
  cloud-provider: "{cloud}"
  cloud-config: "{cloud-config-path}"
apiServerExtraVolumes:
- name: cloud
  hostPath: "{cloud-config-path}"
  mountPath: "{cloud-config-path}"
controllerManagerExtraArgs:
  cloud-provider: "{cloud}"
  cloud-config: "{cloud-config-path}"
controllerManagerExtraVolumes:
- name: cloud
  hostPath: "{cloud-config-path}"
  mountPath: "{cloud-config-path}"
---
```
@kubernetes/sig-cluster-lifecycle-pr-reviews @dims @liztio

### CODE REVIEW

**Comment by dims** [comment]
@luxas please add in the release note that anyone who needs used the `PrivilegedPods ` option should edit the generated files /etc/kubernetes/manifests and add the section `SecurityContext`/`Privileged` sections as necessary

**Comment by dims** [comment]
LGTM 👍

**Comment by luxas** [comment]
Thanks @dims, done :+1:

File: cmd/kubeadm/app/apis/kubeadm/types.go [hunk 1]
```diff
@@ -42,8 +42,6 @@ type MasterConfiguration struct {
 	Networking Networking
 	// KubernetesVersion is the target version of the control plane.
 	KubernetesVersion string
-	// CloudProvider is the name of the cloud provider.
-	CloudProvider string
```
Comment (liztio) [inline cmd/kubeadm/app/apis/kubeadm/types.go]: what's replacing this? Did I miss a discussion?

Comment (liztio) [reply cmd/kubeadm/app/apis/kubeadm/types.go]: Ah, nevermind, gotta read the descriptions first! lgtm

**Comment by seh** [comment]
@luxas, can you clarify the significance of the `apiServerExtraVolumes` and `controllerManagerExtraVolumes` stanzas proposed [above](https://github.com/kubernetes/kubernetes/pull/63866#issue-188147624)? What are those two paths for the two "cloud" volumes? Does the "mount-path" value have to match the "cloud-config" value?

**Comment by luxas** [comment]
@seh is the explanation more clear now? They should all match the cloud config path if you need that.

**Comment by seh** [comment]
Yes, and thank you!
It wasn't clear to me that the file had to be mounted into those containers, because it wasn't clear to me that the API server or controller manager run in containers.

**Pull request merged**